openedx/core/djangoapps/programs/rest_api/v1/views.py

```diff
@@ -110,39 +105,37 @@ def get(self, request, enterprise_uuid):
 
         return Response(programs)
 
-    def _combine_programs_data_and_progress(self, programs_data, programs_progress):
+    def _combine_programs_data_and_progress(
+        self,
+        programs_data: list[dict | None],
+        programs_progress: list[dict | None],
+    ) -> list[dict | None]:
```

Comment:
For my personal edification, I've sometimes seen type hints that are of {some_type} or None that look something like... `List[Optional[some_type]]` but I'm not familiar with this syntax (I don't remember seeing it before). Is this the preferred way to express the variance?

Reply:
oooh, TIL, I was just reading through https://peps.python.org/pep-0604/

Reply:
it appears to be the modern sense. They keep changing the syntax for PEP-484 hint language  with each Python variant. the old syntax still works, though.